File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(long x) {
        if(x<0){
            return false;
        }
       long reverse = 0;
        long temp = x;
        while(temp>0){
            reverse = reverse *10+(temp%10);
            temp = temp/10;
        }
      return reverse==x; 
        
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long long reverse = 0;
        int temp = x;
        while(x!=0){
            long long r = x%10;
            reverse = reverse * 10 + r;
            x = x/10;
        }
        if(reverse != temp || reverse < 0){
            return false;
        }else{
            return true;
        }
    }
};